files/public.go

```diff
@@ -119,6 +119,26 @@ func (storage *PublishedStorage) RemoveDirs(path string, progress aptly.Progress
 	return os.RemoveAll(filepath)
 }
 
+// Cleans up PublishedStorage tree towards the root up to first non-empty directory
+func (storage *PublishedStorage) CleanupPublishTree(path string) {
+	/*
+		Say, `path` is "a/b/c/d", let's remove all empty dirs:
+		 storage.rootPath + "a/b/c/d"
+		 storage.rootPath + "a/b/c"
+		 storage.rootPath + "a/b"
+		 ... and so on, up to first non-empty directory (or any other os.Remove error)
+	*/
+	segments := strings.Split(path, "/")
+
+	for segmentsCount := len(segments); segmentsCount > 0; segmentsCount-- {
+		err := os.Remove(filepath.Join(storage.rootPath, strings.Join(segments[:segmentsCount], "/")))
+
+		if err != nil {
+			break
```

Comment:
please log the error. we canna see if we have werid permissions like 000 or other behavior.

Reply:
Okay, log.Printf will suffice? Or should I bring "github.com/rs/zerolog/log" which is used in other files there?

Reply:
in line 117 progress.Printf is used in RemoveDirs. I think we should follow that pattern...

Reply:
> or return an fmt.Errorf

and what should be done then? It's not the error, that needs any fruther handling.

Implemented it via progress.Printf as you proposed.

Reply:
on a (misconfigured) remote file server, there might be permission issues when removing directories. such errors should be logged... this is why I would still like to have a check for emptyness, bcs then we do not get an error. 

do we have a check where a directory is not removed bcs not empty ?

another little thing: progress logs in aptly do not have punctiations, could you make the log similar to the others ?

Reply:
Oh, too much moving parts here..

> should be logged

but strings printed by aptly.Progress are seen only during cli operation

> remote file server

In api mode all logging is handled by gin or zerolog, I do not know for sure, but plain fmt.Println from gin controller in aptly outputs nothing to stdout.

Your intentions regarding localStorage.CleanupPublishTree() "errors" as you call them is not clear.

Imagine, we will return an error object from it, handle it in `func (collection *PublishedRepoCollection) Remove()` like

```go
		if localStorage, ok := publishedStorage.(*files.PublishedStorage); ok {
			if err := localStorage.CleanupPublishTree(repo.Prefix, progress); err != nil {
				return fmt.Errorf("")
			}
		}
	}

	batch := collection.db.CreateBatch()
	batch.Delete(repo.Key())

	for _, component := range repo.Components() {
		batch.Delete(repo.RefKey(component))
	}

	return batch.Write()
}
```

But is that **really** an error of a scale, that we should terminate the whole process and to not update database at this point?!

Keep in mind, that by the time localStorage.CleanupPublishTree() is executed, all files concerned the publish being `Remove()`-ed are long ago gone. So, if we terminate the execution of Remove() at this point, we'll find ourselves in an inconsistent state!

To my understanding, silently leaving directories intact if we can not remove them is the best bet in this situation. 
Okay, we'll log this in cli via `progress`, but how about api mode?

Also, as I mentioned earlier, I was unable log anything to console by plain printf from gin controller.

Reply:
Notice from `localStorage.CleanupPublishTree` is being logged as per your request and is visible in cli and api modes.